inc/Config/DataSource/HttpDataSource.php

```diff
@@ -98,11 +98,51 @@ protected static function get_service_config_schema(): array {
 		return ConfigSchemas::get_http_data_source_service_config_schema();
 	}
 
+	private static function get_auth_to_request_headers( array $service_config ): array {
+		if ( ! isset( $service_config['auth'] ) ) {
+			return [];
+		}
+
+		$auth_headers = [];
+		$auth = $service_config['auth'];
+
+		if ( 'bearer' === $auth['type'] ) {
+			$auth_headers['Authorization'] = 'Bearer ' . $auth['value'];
+		} elseif ( 'basic' === $auth['type'] ) {
+			$auth_headers['Authorization'] = 'Basic ' . base64_encode( $auth['value'] );
+		} elseif ( 'api-key' === $auth['type'] && 'header' === $auth['add_to'] ) {
+			$auth_headers[ $auth['key'] ] = $auth['value'];
+		}
+
+		return $auth_headers;
```

Comment:
It might be worth adding a filter here to allow customizing these headers. I know we only allow 3 types of authorization schemes, but [there are more schemes possible.](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Authorization). This might be a good way to support those without expanding the dropdown?